### DISCUSSION TITLE
HOW TO ADD FILE IN SOURCE?

### DISCUSSION THREAD

**Discussion by dream811** [obs-websocket-community-projects/obs-websocket-js#246] [Q&A]
I am going to add and delete mp4 file in source 
please share source code for me~~~
Thanks contributors.

**Comment by marckraw** [comment]
Also would like to do that. Is it possibly right now ? for example, by changing the name of a file. In response for getting sources i see only that fields:
![image](https://user-images.githubusercontent.com/8228270/97417830-6a5f5f00-1908-11eb-961b-951e676d776b.png)

so i guess, there is no way to change the filename used in this source ?

**Comment by marckraw** [comment]
I kinda solved my issue, if anyone will have the same. 

You can add source, and then with websockets you can change it's content like that:
```
await obs.send("SetSourceSettings", {
      sourceName: 'your-created-source-name,
      sourceSettings: {
        local_file: "/your/path/to/the/file.mov",
      },
    });
```
